### PR TITLE
Fix/general admin page

### PR DESCRIPTION
**Pull request by Nicolas-Rodriguez-Ch**
### Summary

This pull request contains various features and improvements related to the general admin page.

## Changes Made
- Refactored the privatization of routes to use cookies
- Refactored the user role change for general admin
- Added functionality to delete user and started the refactoring for general admin
- Added English language to the review form
- Added soft deleting to the database and user and restaurant models
- Added create restaurant functionality
- Started with the restaurant deactivation functionality
- Added the update restaurant functionality, images still to be tested
- Added responsive styles to the general admin modal

## Additional Notes

- License and readme were changed in the repository

## To-do
- Test image upload for creation of new restaurant

### CODE REVIEW

File: src/assets/languages/languageSelector.js [hunk 1]
```diff
@@ -240,6 +239,38 @@ const languageSelector = (lang, str) => {
     logOutSuccess: "You've logged out successfully",
     restaurantAdminPage: "Administrate your restaurant",
     appAdminPage: "Administrate the plataform",
+    reviewsTitle: 'Leave your opinion',
+    rating: 'Rating',
+    reviewFormTitle: 'Title',
+    reviewFormComments: 'Coments',
```
Comment (ht1204) [inline src/assets/languages/languageSelector.js]: `Comments`

File: src/components/GeneralAdminModal/GeneralAdminModal.jsx [hunk 2]
```diff
@@ -1,59 +1,54 @@
...
 import './GeneralAdminModal.css'
 import { useSelector } from 'react-redux';
 import languageSelector from '../../assets/languages/languageSelector';
 
-const GeneralAdminModal = ({ onClose, editingItem, checkboxValues, handleRestaurantUpdate }) => {
-  const [title, setTitle] = useState(editingItem.restaurantName);
-  const [location, setLocation] = useState(editingItem.location);
-  const [categories, setCategories] = useState(editingItem.categories);
-  const [errors, setErrors] = useState({});
+const GeneralAdminModal = ({ isOpen, onClose, restaurant, onSave }) => {
```
Comment (ht1204) [inline src/components/GeneralAdminModal/GeneralAdminModal.jsx:6]: Remove this blankspace, please.

File: src/components/HeaderComponent/HeaderComponent.jsx [hunk 3]
```diff
@@ -34,7 +35,7 @@ const HeaderComponent = () => {
     <header className="headerNavBar">
       {showLogIn ? <LogInComponent setShowLogIn={setShowLogIn} /> : ""}
```
Comment (ht1204) [inline src/components/HeaderComponent/HeaderComponent.jsx]: I suggest this:
`{showLogIn ? <LogInComponent setShowLogIn={setShowLogIn} /> : <></>}`

File: src/pages/GeneralAdminView/GeneralAdminUserChange/GeneralAdminUserChange.jsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      }, config);
+      setEmail('');
+      toast.success(response.data.message);
+    } catch (error) {
+      toast.error(error.response?.data?.message || languageSelector(language, 'updateUserError'));
+    }
+  }
+  
+  const hanldeUserDelete = async (id) => {
+    if (window.confirm(languageSelector(language, 'deavtivateUserWarning'))) {
```
Comment (ht1204) [inline src/pages/GeneralAdminView/GeneralAdminUserChange/GeneralAdminUserChange.jsx]: `deactivateUserWarning`

File: src/pages/GeneralAdminView/NewRestaurantForm/NewRestaurantForm.jsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    if (!restaurant_name.trim()) {
+      validationErrors.restaurant_name = languageSelector(language, 'newRestaurantFormNameError');
+    }
+    if (!logo) {
+      validationErrors.logo = languageSelector(language, 'newRestaurantLogoError');
+    }
+    if (!mainPhoto) {
+      validationErrors.mainPhoto = languageSelector(language, 'newRestaurantMainPhotoError');;
+    }
+    if (!adminEmail.trim() || !/^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/.test(adminEmail)) {
```
Comment (ht1204) [inline src/pages/GeneralAdminView/NewRestaurantForm/NewRestaurantForm.jsx]: Move this regex to a constant to be imported through the project

File: src/pages/GeneralAdminView/GeneralAdminUserChange/GeneralAdminUserChange.jsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  }, []);
+  
+  const toggleExpanded = () => {
+    setIsExpanded(!isExpanded);
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      const response = await axios.put(`${process.env.REACT_APP_API_URL}/users/change-role`, {
```
Comment (ht1204) [inline src/pages/GeneralAdminView/GeneralAdminUserChange/GeneralAdminUserChange.jsx]: I consider API_URL should be imported using a constant through the project. CC @dgaleano201611

Comment (Nicolas-Rodriguez-Ch) [reply src/pages/GeneralAdminView/GeneralAdminUserChange/GeneralAdminUserChange.jsx]: Resolved

File: src/pages/GeneralAdminView/NewRestaurantForm/NewRestaurantForm.jsx (hunk 5 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/pages/GeneralAdminView/NewRestaurantForm/NewRestaurantForm.jsx]: Resolved

File: src/pages/GeneralAdminView/GeneralAdminUserChange/GeneralAdminUserChange.jsx (hunk 4 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/pages/GeneralAdminView/GeneralAdminUserChange/GeneralAdminUserChange.jsx]: Resolved

File: src/components/HeaderComponent/HeaderComponent.jsx (hunk 3 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/HeaderComponent/HeaderComponent.jsx]: Resolved

File: src/components/GeneralAdminModal/GeneralAdminModal.jsx (hunk 2 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/GeneralAdminModal/GeneralAdminModal.jsx:6]: Resolved

File: src/assets/languages/languageSelector.js (hunk 1 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/assets/languages/languageSelector.js]: Resolved

**Comment by Nicolas-Rodriguez-Ch** [comment]
New commit that addresses the feedback

**Review by ht1204 — APPROVED** [review]

**Pull request merged**